x-pack/solutions/security/plugins/elastic_assistant/server/ai_assistant_data_clients/find.ts

```diff
@@ -72,7 +72,9 @@ export const findDocuments = async <TSearchSchema>({
         query,
         track_total_hits: true,
         sort,
-        _source: true,
+        _source: {
+          includes: fields,
+        },
```

Comment:
```suggestion
        _source: fields.length ? {
          includes: fields,
        } : true,
```

Reply:
Sorry I may have approved too soon...
```suggestion
        _source: fields?.length ? false : true,
        ...(fields?.length ? { fields: fields.map((f) => snakeCase(f)) } : {}),
```